src/client.ts

```diff
@@ -583,12 +589,24 @@ export class VocdoniSDKClient {
    * @returns {Promise<string>} Resulting election id.
    */
   async createElection(election: UnpublishedElection): Promise<string> {
+    invariant(
+      election.maxCensusSize || election.census.type !== CensusType.CSP,
+      'CSP Census needs a max census size set in the election'
+    );
+
+    const chainId = await this.fetchChainId();
+
     if (!election.census.isPublished) {
       await this.createCensus(election.census as PlainCensus | WeightedCensus);
+    } else if (!election.maxCensusSize) {
+      await this.fetchCensusInfo(election.census.censusId).then((censusInfo) => {
+        election.census.size = censusInfo.size;
+        election.census.weight = censusInfo.weight;
```

Comment:
@marcvelmer should `election.maxCensusSize` be set in this case?

Reply:
No, the `election.maxCensusSize` is for the user to be set and it has no relation with the real census size. In case there is no `election.maxCensusSize` set, then we use the real size of the census, but just because now it's mandatory to have this value.